### ISSUE TITLE
Mismatch in audio frames returned by pyav and video reader

### ISSUE THREAD

**Issue by prabhat00155** [pytorch/vision#3986]
[audio-video sync test](https://github.com/pytorch/vision/blob/964ce1e95cba640768c5a39dbc018531c49d476c/test/test_video_reader.py#L1238) was failing in https://github.com/pytorch/vision/pull/3934. However, it turns out that the audio frames returned by our video reader doesn't match with pyav results(both output shapes and values are different).
Here is the [notebook](https://colab.research.google.com/drive/1G1_ejPJgEB-ljaAanBYnvlE02u-8ZWlb?usp=sharing) with details.

**Comment by mthrok** [comment]
**Off-topic, yet remotely related** Does the video reader handle stereo sound as well?
When testing something with audio, I recommend to use stereo audio just to be sure.

**Comment by prabhat00155** [comment]
> _[Off-topic, yet remotely related]_ Does the video reader handle stereo sound as well?
> When testing something with audio, I recommend to use stereo audio just to be sure.

[read_video](https://github.com/pytorch/vision/blob/0013d9314cf1bd83eaf38c3ac6e0e9342fa99683/torchvision/io/video.py#L256) and [write_video](https://github.com/pytorch/vision/blob/0013d9314cf1bd83eaf38c3ac6e0e9342fa99683/torchvision/io/video.py#L111) support audio frames with multiple channels, so I would think yes. @bjuncek can confirm this.

**Comment by bjuncek** [comment]
Yes, we _do_ handle it in normal circumstances. I've learned over the course of debugging this one that there might be more than one way of encoding stereo audio, so I have to double check if we're good for all the edge cases.
